app/controllers/restaurants_controller.rb

```diff
@@ -24,6 +24,20 @@ def create
     end
   end
 
+  def edit
+    @restaurant = Restaurant.find(params[:id])
+  end
+
+  def update
+    @restaurant = Restaurant.find(params[:id])
+    if @restaurant.update_attributes(restaurant_params)
+      flash[:success] = 'Changes saved successfully'
+      redirect_to restaurant_path(@restaurant)
+    else
+      render :edit
```

Comment:
It would be good to flash an error message if the restaurant doesn't successfully save